v2/ical.NET/Serialization/SimpleDeserializer.cs

```diff
@@ -0,0 +1,221 @@
+            }
+        }
+
+        private CalendarProperty ParseContentLine(SerializationContext context, string input)
+        {
+            var match = _contentLineRegex.Match(input);
+            if (!match.Success)
+            {
+                throw new SerializationException($"Could not parse line: '{input}'");
+            }
+            var name = match.Groups[_nameGroup].Value;
+            var value = match.Groups[_valueGroup].Value;
+            var paramNames = match.Groups[_paramNameGroup].Captures;
+            var paramValues = match.Groups[_paramValueGroup].Captures;
+
+            var property = new CalendarProperty(name.ToUpperInvariant());
+            context.Push(property);
+            SetPropertyParameters(property, paramNames, paramValues);
+            SetPropertyValue(context, property, value);
+            context.Pop();
+            return property;
+        }
+
+        private static void SetPropertyParameters(CalendarProperty property, CaptureCollection paramNames, CaptureCollection paramValues)
+        {
+            var paramValueIndex = 0;
+            for (var paramNameIndex = 0; paramNameIndex < paramNames.Count; paramNameIndex++)
+            {
+                var paramName = paramNames[paramNameIndex].Value;
+                var parameter = new CalendarParameter(paramName);
+                var nextParamIndex = paramNameIndex + 1 < paramNames.Count ? paramNames[paramNameIndex + 1].Index : int.MaxValue;
+                while (paramValueIndex < paramValues.Count && paramValues[paramValueIndex].Index < nextParamIndex)
+                {
+                    var paramValue = paramValues[paramValueIndex].Value;
+                    parameter.AddValue(paramValue);
+                    paramValueIndex++;
+                }
+                property.AddParameter(parameter);
+            }
+        }
+
+        private void SetPropertyValue(SerializationContext context, CalendarProperty property, string value)
+        {
+            var type = _dataTypeMapper.GetPropertyMapping(property) ?? typeof(string);
+            var serializer = (SerializerBase)_serializerFactory.Build(type, context);
+            using (var valueReader = new StringReader(value))
+            {
+                var propertyValue = serializer.Deserialize(valueReader);
+                var propertyValues = propertyValue as IEnumerable<string>;
+                if (propertyValues != null)
+                {
+                    foreach (var singlePropertyValue in propertyValues)
+                    {
+                        property.AddValue(singlePropertyValue);
+                    }
+                }
+                else
+                {
+                    property.AddValue(propertyValue);
+                }
+            }
+        }
+
+        private static IEnumerable<string> GetContentLines(TextReader reader)
+        {
+            var currentLine = new StringBuilder();
```

Comment:
Yeah, I suspect my obsession with truly micro micro-optimizations only matters in a context like high-frequency trading where nanoseconds matter. Or if you're writing Kestrel or [RavenDb](https://ayende.com/blog/175713/timing-the-time-it-takes-to-parse-time-part-i).